File: ncarg2d/src/libncarg/plotchar/pcpnwi.f
C
C $Id: pcpnwi.f,v 1.7 2008-07-27 00:17:20 haley Exp $
C                                                                      
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      CHARACTER*16 FUNCTION PCPNWI (WHCH,IPAI)
C
      CHARACTER*(*) WHCH
      CHARACTER*16   CTMP
C
C Given the name (WHCH) of an internal parameter that is really an array
C and the integer value (IPAI) of the desired parameter array index, the
C value of PCPNWI (WHCH,IPAI) is the internal parameter name including
C the index, in the form expected by the routines PCGETR and PCSETR (in
C parentheses following the two-character internal parameter name).
C
      IF (LEN(WHCH).LT.2) THEN
        CALL SETER ('PCPNWI - PARAMETER NAME TOO SHORT',1,1)
        RETURN
      ELSE
        WRITE (CTMP,'(I16)') IPAI
        IBEG=0
        DO 101 I=1,16
          IF (CTMP(I:I).NE.' ') THEN
            IF (IBEG.EQ.0) IBEG=I
            IF (IBEG.NE.0) IEND=I
          END IF
  101   CONTINUE
        IF (IBEG.EQ.0) THEN
          CALL SETER ('PCPNWI - INTERNAL ERROR - SEE CONSULTANT',2,1)
          RETURN
        ELSE
          PCPNWI=WHCH(1:2)//'('//CTMP(IBEG:IEND)//')'//'            '
        END IF
      END IF
C
C Done.
C
      RETURN
C
      END

File: ncarg2d/man/plotchar/pcpnwi.m
.TH PCPNWI 3NCARG "March 1993" UNIX "NCAR GRAPHICS"
.na
.nh
.SH NAME
PCPNWI - A function of type CHARACTER*16.
.SH SYNOPSIS
.nf
CHARACTER*16 PCPNWI,ENAM
ENAM = PCPNWI (PNAM,IPAI)
.fi
.SH C-BINDING SYNOPSIS
#include <ncarg/ncargC.h>
.sp
void c_pcpnwi (char *pnam, int *ipai)
.SH DESCRIPTION 
.IP PNAM 12
(an input constant or variable of type CHARACTER) is a character string,
the first two characters of
which are the name of an internal parameter array of
Plotchar (currently, \'BC\' or \'CC\').
.IP IPAI 12
(an input expression of type INTEGER)
is an integer expression specifying the index of that
element of the internal parameter array of which one wants
the name.
.PP
Given the above two arguments, the value of PCPNWI(PNAM,IPAI) is that
character string which serves as the name of element IPAI of the internal
parameter array identified by PNAM.
.sp
For example, if IPAI has the value 3, then the value of
"PCPNWI('CC',IPAI)" is the character string 'CC(3)' (padded with blanks
to the right) and the call "CALL PCSETI (PCPNWI('CC',IPAI),62)" would
give the value 62 to the third element of the parameter array 'CC'.
.SH C-BINDING DESCRIPTION
The C-binding argument descriptions are the same as the FORTRAN
argument descriptions.
.SH USAGE
This routine generates parameter names for use in calls to
other routines. 
For a complete list of parameters available
in this utility, see the plotchar_params man page.
.SH ACCESS
To use PCPNWI or c_pcpnwi, load the NCAR Graphics libraries ncarg, ncarg_gks,
and ncarg_c, preferably in that order.  
.SH MESSAGES
See the plotchar page for a description of all Plotchar error 
messages and/or informational messages.
.SH SEE ALSO
Online:
plotchar,
plotchar_params,
pcdlsc,
pcgetc,
pcgeti,
pcgetr,
pchiqu,
pcloqu,
pcmequ,
pcmpxy,
pcrset,
pcsetc,
pcseti,
pcsetr,
ncarg_cbind.
.sp
Hardcopy:
NCAR Graphics Fundamentals, UNIX Version
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br
The use of this Software is governed by a License Agreement.
